abel/direct.py

```diff
@@ -209,28 +205,28 @@ def _pyabel_direct_integral(f, r, correction, int_func=np.trapz):
 
 
     """
-    Compute the correction
-    Pre-calculated analytical integration of the cell with the singular value
-    Assuming a piecewise linear behaviour of the data
+    Compute the correction. Here we apply an
+    analytical integration of the cell with the singular value,
+    assuming a piecewise linear behaviour of the data.
+    The analytical abel transform for this trapezoid is:
     c0*acosh(r1/y) - c_r*y*acosh(r1/y) + c_r*sqrt(r1**2 - y**2)
+    see: https://github.com/luli/hedp/blob/master/hedp/math/abel.py#L87-L104
     """
     if correction == 1:
-        # computing forward derivative of the data
-        f_r = (f[:, 1:] - f[:, :N1-1])/np.diff(r)[None, :]
```

Comment:
No need for `N1` it and its assignment may be deleted.  `f[:, -1]` is equally as good.